presto-hive/src/main/java/com/facebook/presto/twitter/hive/thrift/ThriftHiveRecordCursorProvider.java

```diff
@@ -92,6 +92,8 @@ public ThriftHiveRecordCursorProvider(HdfsEnvironment hdfsEnvironment, ThriftFie
 
         return Optional.of(new ThriftHiveRecordCursor<>(
                 genericRecordReader(recordReader),
+                path,
+                start,
```

Comment:
Just pure personal opinion: seems these two params are passed in just for a more detailed exception message. I personally prefer not to mix in anything if they aren't needed logically.
What do you think of just try-catch in CursorProvider and throw the detailed exception here?

Also I think if we just go with String.format we avoid having to add a line for import. :)